workflow-runtime/build.gradle.kts

```diff
@@ -18,35 +14,51 @@ java {
 
 apply(from = rootProject.file(".buildscript/configure-maven-publish.gradle"))
 
-// Benchmark configuration.
-configure<JMHPluginExtension> {
-  include = listOf(".*")
-  duplicateClassesStrategy = DuplicatesStrategy.WARN
-}
-configurations.named("jmh") {
-  attributes.attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage.JAVA_RUNTIME))
-}
-tasks.named<KotlinCompile>("compileJmhKotlin") {
-  kotlinOptions {
-    // Give the benchmark code access to internal definitions.
-    val compileKotlin: KotlinCompile by tasks
-    freeCompilerArgs += "-Xfriend-paths=${compileKotlin.destinationDir}"
+kotlin {
+  jvm {
+    compilations {
+      val main by getting
+      val workflowNode by creating {
+        kotlinOptions {
+          val compileKotlinJvm: KotlinCompile by tasks
+          freeCompilerArgs += "-Xfriend-paths=${compileKotlinJvm.destinationDir}"
```

Comment:
If we don't think friend-path arguments work wiht the multiplatform plugin, why do we still have this here?

Reply:
I looked into this a bit.  `friend-paths` is working here, and it's necessary.  In order for it to work, we also need the dependency upon `main.output.classesDirs` which is added below.  Why is that necessary?  No idea.

We can technically kind of do the same thing with the testing module, but of course that's a published artifact, and it's pointing to a different published artifact.  We're pointing Gradle at actual files and I can't imagine how it would work using Maven coordinates.

Reply:
Managed to get friend paths working. Raised another PR.